dataclients/kube/kube.go

```diff
@@ -0,0 +1,303 @@
+package kube
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"regexp"
+	"strings"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/zalando/skipper/eskip"
+)
+
+// FEATURE:
+// - provide option to limit the used namespaces?
+
+const (
+	defaultAPIAddress = "http://localhost:8001"
```

Comment:
Maybe it's a premature optimization, but I would make it possible to configure API access based on the environment when running in a cluster, so you don't have to have a 80+ MiB sidecar with `kubectl` just for this.

Example of how it's done in the kubernetes go client: https://github.com/kubernetes/client-go/blob/master/rest/config.go#L270-L298

I aslo made a PR for this in mate: https://github.com/zalando-incubator/mate/pull/21 :)